src/profile-logic/import/instruments/index.js

```diff
@@ -0,0 +1,899 @@
+      stackKeyToIndex,
+      rootStackKey,
+      null,
+      rootFrameIndex
+    );
+  }
+
+  for (const frameData of addressToFrameMap) {
+    const frameMetaData = frameData[1];
+    const frameAddress = frameData[0];
+
+    const indexToFunc = getOrCreateFunc(
+      funcTable,
+      funcKeyToIndex,
+      stringTable,
+      frameMetaData.name,
+      frameMetaData.file || '',
+      frameMetaData.key + ''
+    );
+
+    createFrame(
+      frameTable,
+      stringTable,
+      frameKeyToIndex,
+      indexToFunc,
+      frameAddress + ''
+    );
+  }
+
+  for (const sample of samples) {
+    const stackTrace = sample.backtraceStack;
+    let parentIndex = stackKeyToIndex.get(rootStackKey);
+    if (stackTrace) {
+      for (let index = 0; index < stackTrace.length; index++) {
+        const frameAddress = stackTrace[index];
+        const keyOfStackKeyToIndexMap =
+          typeof parentIndex === 'number'
+            ? '$' + parentIndex + '$' + frameAddress
+            : '$' + frameAddress;
+
+        if (!stackKeyToIndex.has(keyOfStackKeyToIndexMap)) {
+          const frameIndex = frameKeyToIndex.get(frameAddress + '');
+
+          if (
+            typeof frameIndex === 'number' &&
+            typeof parentIndex === 'number'
+          ) {
+            createStack(
+              stackTable,
+              stringTable,
+              stackKeyToIndex,
+              keyOfStackKeyToIndexMap,
+              parentIndex,
+              frameIndex
+            );
+          }
+        }
+
+        parentIndex = stackKeyToIndex.get(keyOfStackKeyToIndexMap);
+
+        if (index === stackTrace.length - 1) {
+          const stackForSample = stackKeyToIndex.get(keyOfStackKeyToIndexMap);
+          if (stackForSample) {
+            samplesTable.stack.push(stackForSample);
+          } else {
+            samplesTable.stack.push(null);
+          }
+          samplesTable.time.push(sample.timestamp / 1000000);
+          samplesTable.responsiveness.push(null);
+          samplesTable.length++;
+        }
+      }
+    }
+  }
+
+  return thread;
+}
+
+// This function creates a thread for each group of samples(by threadID)
+function pushThreadsInProfile(profile, addressToFrameMap, samples) {
+  const threadIDToSamples = new Map();
+  for (const sample of samples) {
+    const samplesArray = threadIDToSamples.get(sample.threadID);
+    if (samplesArray) {
+      samplesArray.push(sample);
+    } else {
+      threadIDToSamples.set(sample.threadID, [sample]);
+    }
+  }
+
+  for (const threadID of threadIDToSamples.keys()) {
+    const samples = threadIDToSamples.get(threadID);
+    let processedThread = getEmptyThread();
+    if (samples) {
+      processedThread = getProcessedThread(
+        threadID,
+        samples,
+        addressToFrameMap
+      );
+    }
+    profile.threads.push(processedThread);
+  }
+}
+
+export async function convertInstrumentsProfile(
+  entry: { name: string },
+  fileReaderHelper: fileReader
```

Comment:
I didn't understand why we use `fileReader` as the type but at least this needs to be `typeof fileReader`.